app/main.py

```diff
@@ -1,2 +1,49 @@
+class Node:
+    def __init__(self, key: object, value: object) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 16, load_factor: float = 0.75) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.load_factor = load_factor
+        self.table = [[] for _ in range(capacity)]
+
+    def _resize(self) -> None:
+        old_table = self.table
+        self.capacity *= 2
+        self.table = [[] for _ in range(self.capacity)]
+        self.size = 0
+        for bucket in old_table:
+            for node in bucket:
+                self[node.key] = node.value
+
+    def _hash_index(self, key: object) -> int:
+        return hash(key) % self.capacity
+
+    def __setitem__(self, key: object, value: object) -> None:
```

Comment:
Key must be a hashable object, value can be anything - fix this annotation in all methods where applicable